examples/z_pull.c

```diff
@@ -23,10 +23,10 @@
 const char *kind_to_str(z_sample_kind_t kind);
 
 void data_handler(const z_sample_t *sample, void *arg) {
-    char *keystr = z_keyexpr_to_string(sample->keyexpr);
-    printf(">> [Subscriber] Received %s ('%s': '%.*s')\n", kind_to_str(sample->kind), keystr, (int)sample->payload.len,
-           sample->payload.start);
-    free(keystr);
+    z_owned_str_t keystr = z_keyexpr_to_string(sample->keyexpr);
+    printf(">> [Subscriber] Received %s ('%s': '%.*s')\n", kind_to_str(sample->kind), z_loan(keystr),
+           (int)sample->payload.len, sample->payload.start);
+    z_drop(&keystr);
```

Comment:
In Zenoh-Pico, we have it implemented as `z_drop(z_move(keystr))`.
I know it might seem redundant but by doing so we dont break the rule of being explicit when moving things to the Zenoh API.